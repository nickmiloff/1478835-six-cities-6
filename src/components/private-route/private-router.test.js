import React from 'react';
import {Route} from 'react-router-dom';
import {render} from '@testing-library/react';
import Test from '../test/test';

import PrivateRoute from './private-route';

describe(`PrivateRoute component test`, () => {
  const MockComponent = () => <h1>Mock component</h1>;

  it(`'PrivateRoute component' should render correctly 'component' with authorized user`, () => {
    const {container} = render(
        <Test initial={false}>
          <Route exact path="/login"><h1>Login page</h1></Route>
          <PrivateRoute component={MockComponent} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`'PrivateRoute component' should correctly redirect to '/' with unauthorized user`, () => {
    const {container} = render(
        <Test>
          <Route exact path="/login"><h1>Login page</h1></Route>
          <PrivateRoute component={MockComponent} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
