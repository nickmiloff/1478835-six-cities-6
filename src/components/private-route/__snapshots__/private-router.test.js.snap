// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component test 'PrivateRoute component' should correctly redirect to '/' with unauthorized user 1`] = `
<div>
  <h1>
    Login page
  </h1>
</div>
`;

exports[`PrivateRoute component test 'PrivateRoute component' should render correctly 'component' with authorized user 1`] = `
<div>
  <h1>
    Mock component
  </h1>
</div>
`;
